mutable_buffer/src/table.rs

```diff
@@ -205,31 +201,38 @@ impl Table {
         }
 
         // make sure all the columns are of the same length
-        for col in &mut self.columns {
+        for col in self.columns.values_mut() {
             col.push_none_if_len_equal(row_count);
         }
 
         Ok(())
     }
 
     pub fn row_count(&self) -> usize {
-        self.columns.first().map_or(0, |v| v.len())
+        if let Some(col) = self.columns.values().next() {
+            return col.len();
+        }
+
+        0
```

Comment:
If you wanted to get all rusty you could maybe do something like 

```suggestion
        self.columns.values().next()
            .map(|col| col.len())
            .unwrap_or(0)
```

Though I can't see how it really matters one way or the other

Reply:
Ah, that's a little cleaner, will update